api/tools/src/main/java/org/pmiops/workbench/tools/SetAuthority.java

```diff
@@ -45,19 +45,30 @@ public CommandLineRunner run(UserDao userDao) {
     return (args) -> {
       // User-friendly command-line parsing is done in devstart.rb, so we do only simple positional
       // argument parsing here.
-      if (args.length != 4) {
+      if (args.length != 5) {
         throw new IllegalArgumentException(
-            "Expected 4 args (email_list, authorities, remove, dry_run). Got "
+            "Expected 5 args (email_list, authorities, remove, dry_run, confirm_add_all). Got "
                 + Arrays.asList(args));
```

Comment:
When I run `./gradlew setAuthority` (note: no arguments) this fails. However, I cannot see the error message. I have the same problem with all our project.rb-derived tooling, which is why I added a "Severe" log message below.

Is anyone else able to see this? If so, do I have something configured incorrectly?

Reply:
When I try this, I see
```
2023-05-01 16:25:24.555  WARN 43487 --- [           main] s.c.a.AnnotationConfigApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'dataSourceScriptDatabaseInitializer' defined in class path resource [org/springframework/boot/autoconfigure/sql/init/DataSourceInitializationConfiguration.class]: Unsatisfied dependency expressed through method 'dataSourceScriptDatabaseInitializer' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.pmiops.workbench.db.WorkbenchDbConfig': Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.pmiops.workbench.db.Params': Instantiation of bean failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.pmiops.workbench.db.Params$$EnhancerBySpringCGLIB$$9683b66b]: Constructor threw exception; nested exception is java.lang.IllegalStateException: Database connection requires either a hostname (DB_HOST) or a Cloud SQL instance name (CLOUD_SQL_INSTANCE_NAME).
```

I'm guessing that is not what you mean?

In my case it seems to be failing before it even has a chance to parse args

Reply:
The example I used wasn't the exact command I ran. Sorry about that!

Try changing the ruby code in `set_authority` to the following:

```
ServiceAccountContext.new(gcc.project).run do
  Common.new.run_inline %W{
    ./gradlew setAuthority
   -PappArgs=['#{op.opts.email}','#{op.opts.authority}']} # less arguments
end
```

Reply:
Ah ok.  With that change, I also see an error but no explanation.
```
thibault@wm242-74d ~/src/aou/api (peterlavigne/rw-9958-user-input-v2) $ ./project.rb set-authority --project  all-of-us-workbench-test --email foo --authority bar
Reading gcloud configuration...
  account: joel.thibault@pmi-ops.org
Attaching to existing keyfile @ /Users/thibault/src/aou/api/sa-key.json
+ ./gradlew setAuthority -PappArgs=['foo','bar']

> Task :setAuthority

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::                (v2.7.4)
2023-05-01 16:53:47.783  INFO 44605 --- [           main] .t.CommandLineToolApplicationRunListener : Registering fake REQUEST scope to the context bean factory for command-line tools
2023-05-01 16:53:47.786  INFO 44605 --- [           main] org.pmiops.workbench.tools.SetAuthority  : Starting SetAuthority using Java 11.0.18 on wm242-74d with PID 44605 (/Users/thibault/src/aou/api/build/classes/java/__tools__ started by thibault in /Users/thibault/src/aou/api)
2023-05-01 16:53:47.786  INFO 44605 --- [           main] org.pmiops.workbench.tools.SetAuthority  : No active profile set, falling back to 1 default profile: "default"
2023-05-01 16:53:48.084  INFO 44605 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2023-05-01 16:53:48.189  INFO 44605 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 100 ms. Found 32 JPA repository interfaces.
2023-05-01 16:53:48.202  INFO 44605 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2023-05-01 16:53:48.250  INFO 44605 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 47 ms. Found 32 JPA repository interfaces.
2023-05-01 16:53:48.455  INFO 44605 --- [           main] org.pmiops.workbench.db.Params           : Workbench SQL instance params: [hostname:null cloudSqlInstanceName:all-of-us-workbench-test:us-central1:workbenchmaindb username:workbench password:p5*****************]
2023-05-01 16:53:48.466  INFO 44605 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2023-05-01 16:53:48.525  INFO 44605 --- [           main] c.g.cloud.sql.core.CoreSocketFactory     : Connecting to Cloud SQL instance [all-of-us-workbench-test:us-central1:workbenchmaindb] via SSL socket.
2023-05-01 16:53:48.525  INFO 44605 --- [           main] c.g.cloud.sql.core.CoreSocketFactory     : First Cloud SQL connection, generating RSA key pair.
2023-05-01 16:53:50.281  INFO 44605 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
2023-05-01 16:53:50.357  INFO 44605 --- [           main] o.hibernate.jpa.internal.util.LogHelper  : HHH000204: Processing PersistenceUnitInfo [name: workbench]
2023-05-01 16:53:50.387  INFO 44605 --- [onnection adder] c.g.cloud.sql.core.CoreSocketFactory     : Connecting to Cloud SQL instance [all-of-us-workbench-test:us-central1:workbenchmaindb] via SSL socket.
2023-05-01 16:53:50.391  INFO 44605 --- [           main] org.hibernate.Version                    : HHH000412: Hibernate ORM core version 5.6.9.Final
2023-05-01 16:53:50.493  INFO 44605 --- [           main] o.hibernate.annotations.common.Version   : HCANN000001: Hibernate Commons Annotations {5.1.2.Final}
2023-05-01 16:53:50.602  INFO 44605 --- [           main] org.hibernate.dialect.Dialect            : HHH000400: Using dialect: org.hibernate.dialect.MySQL57Dialect
2023-05-01 16:53:51.058  INFO 44605 --- [onnection adder] c.g.cloud.sql.core.CoreSocketFactory     : Connecting to Cloud SQL instance [all-of-us-workbench-test:us-central1:workbenchmaindb] via SSL socket.
2023-05-01 16:53:51.320  INFO 44605 --- [           main] o.h.e.t.j.p.i.JtaPlatformInitiator       : HHH000490: Using JtaPlatform implementation: [org.hibernate.engine.transaction.jta.platform.internal.NoJtaPlatform]
2023-05-01 16:53:51.325  INFO 44605 --- [           main] j.LocalContainerEntityManagerFactoryBean : Initialized JPA EntityManagerFactory for persistence unit 'workbench'
2023-05-01 16:53:51.369  INFO 44605 --- [           main] org.javers.core.JaversBuilder            : mappingStyle: FIELD
2023-05-01 16:53:51.370  INFO 44605 --- [           main] org.javers.core.JaversBuilder            : loading GuavaAddOns ...
2023-05-01 16:53:51.400  INFO 44605 --- [           main] org.javers.core.JaversBuilder            : using fake InMemoryRepository, register actual Repository implementation via JaversBuilder.registerJaversRepository()
2023-05-01 16:53:51.410  INFO 44605 --- [           main] org.javers.core.JaversBuilder            : JaVers instance started in 62 ms
2023-05-01 16:53:51.631  INFO 44605 --- [onnection adder] c.g.cloud.sql.core.CoreSocketFactory     : Connecting to Cloud SQL instance [all-of-us-workbench-test:us-central1:workbenchmaindb] via SSL socket.
2023-05-01 16:53:52.150  INFO 44605 --- [           main] org.pmiops.workbench.tools.SetAuthority  : Started SetAuthority in 4.63 seconds (JVM running for 4.955)
2023-05-01 16:53:52.169  INFO 44605 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown initiated...
2023-05-01 16:53:52.790  INFO 44605 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown completed.

> Task :setAuthority FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':setAuthority'.
> Process 'command '/Library/Java/JavaVirtualMachines/temurin-11.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 7s
21 actionable tasks: 1 executed, 20 up-to-date
thibault@wm242-74d ~/src/aou/api (peterlavigne/rw-9958-user-input-v2) $
```

Reply:
Thanks. So it's not just my environment, good to know.